Prevent IcDataTable action element clicks from selecting the table row

dn55533 opened the pull request:
## Summary of the changes
Provide access to the the click event in the Data Table action element, thereby allowing the event propagation to be stopped in individual data table action element click handlers.

This is to prevent IcDataTable rows from becoming selected when the row contains an IcButton in an action element and the button is clicked.

![IcButton prevent propagation](https://github.com/user-attachments/assets/dd85eea5-5ca7-4066-aba0-0efd23ca6b5f)

## Related issue
N/A

## Checklist

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

ad9242 reviewed (changes requested):
This seems to have introduced some problems. For example: go to React Storybook : Dialog > Sizes > Launch Medium Dialog. Clicking the show button has no effect:

![image](https://github.com/user-attachments/assets/d99071eb-1125-439d-96a2-4ee7017db596)

you can see the intended behaviour on the develop branch:
https://mi6.github.io/ic-ui-kit/branches/develop/react/?path=/story/react-components-dialog--sizes

I think it may have also introduced other problems that are being caught by the cypress tests in the CI pipeline

I think it would be best to add this new functionality under a prop (defaulting to false)

dn55533 commented:
I have removed the call to stop propagation from the IcButton component because this was causing various issues. Instead I have exposed the click event in the Data Table action element, thereby allowing the event propagation to be stopped in individual data table action element click handlers.

I have updated the stories and example code accordingly.

ad9242 reviewed (changes requested):
It would be good to add a visual regression test for this. Whilst it's kind of on the developer to add the stopPropogation call into the actionElementOnClick, it would mean we don't risk breaking anything in future if there are changes in the core data-table

ad9242 reviewed (changes requested):
sorry, I know I said a VR test, but to save the issue with generating the image, it could just check for the class name:

```
const TABLE_ROW_SELECTED = "table-row-selected";

it("should not highlight the selected row when the action element click event is stopped from propagating", () => {
    mount(
      <IcDataTable
        columns={COLS}
        data={ACTION_DATA_ELEMENTS}
        caption="Data tables"
      ></IcDataTable>
    );

    cy.checkHydrated(DATA_TABLE_SELECTOR);

    cy.get(DATA_TABLE_SELECTOR)
      .shadow()
      .find(`[data-testid="copy-button"]`)
      .eq(0)
      .click();

    cy.findShadowEl(DATA_TABLE_SELECTOR, "tr").eq(1).should(NOT_HAVE_CLASS, TABLE_ROW_SELECTED);

    cy.get(DATA_TABLE_SELECTOR)
      .shadow()
      .find(`[data-testid="copy-button"]`)
      .eq(1)
      .click();
    
    cy.findShadowEl(DATA_TABLE_SELECTOR, "tr").eq(2).should(HAVE_CLASS, TABLE_ROW_SELECTED);
  });
```

Would you mind changing it to this, and remove the added image please? Otherwise happy to approve
